### PR TITLE
Deploy to fly, other updates

### PR DESCRIPTION
**Pull request by sethetter**
- Set up fly.toml config for deploying to fly.io
- Update the `Dockerfile` to expose port 3000
- Add a `.dockerignore` to keep certain files out of the image
- Add a `requireMention` flag to command handlers
- Some type reworking to match what is passed into the `app.message` handler middleware
- Ensure the slack client is being started properly, with command handlers registered
- Rename `KeeperSlackMiddleware` to just `MessageEvent`

[Issue](https://github.com/devict/keeper/issues/1#issue-1211524035)

### CODE REVIEW

File: fly.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# fly.toml file generated for keeper on 2022-05-02T06:36:51-05:00
+
+app = "keeper"
+
+kill_signal = "SIGINT"
+kill_timeout = 5
+processes = []
+
+[env]
+  CONF_FILE = "main"
```
Comment (KalebMills) [inline fly.toml:10]: Just curious, how does the config file actually end up getting its properties? Are the secrets added in the fly console or something?

File: src/lib/clients/slack.ts [hunk 2]
```diff
@@ -22,35 +21,34 @@ class SlackClient implements CommandClient {
...
-            }));
+                this.logger.log(`message received! ${JSON.stringify(args)}`);
+                await handler({ client: this, logger: this.logger, command: args });
+            });
         });
-        this._client.error((error: Error) => {
-            this.logger.error(error)
-            return Promise.resolve();
-        })
+        this._client.error(async (error: Error) => this.logger.error(error));
```
Comment (KalebMills) [inline src/lib/clients/slack.ts]:
```suggestion
        this._client.error(this.logger.error);
```

**Review by KalebMills — APPROVED** [review]
Looks great @sethetter! Great work!

File: fly.toml (hunk 1 above)
Comment (sethetter) [reply fly.toml:10]: I set them with `flyctl secrets set KEY=value`, but the fly web console works as well I believe

**Pull request merged**